service_configuration_lib/spark_config.py

```diff
@@ -292,6 +293,10 @@ def _adjust_spark_requested_resources(
             str(min(executor_instances, DEFAULT_K8S_BATCH_SIZE)),
         )
         user_spark_opts.setdefault('spark.kubernetes.executor.limit.cores', str(executor_cores))
+        user_spark_opts.setdefault(
+            'spark.scheduler.maxRegisteredResourcesWaitingTime',
+            str(10 + executor_instances * DEFAULT_RESOURCES_WAITING_TIME_PER_EXECUTOR // 60) + 'min',
```

Comment:
In my experiments I have seen 15 min. But I think 10 is also ok, as it's a soft limit.

Reply:
Question: If the user is providing this config, does it overrides it?
Maybe we should take higher of the two, if the user specifies too?

Reply:
> It might be misleading, no? It's not generally the approach that we follow for other settings in this library.
I am not sure about this. 
Example case: For some reason, the calculated waiting time is not enough.
User wants to specify config to higher value, as it's important for them to get the desired executor count.
They won't be able specify it, right? As it will be overridden everytime.
This makes the waiting-time config option non-configurable. 

It seems to me a trivial change to pick-up higher of given and calculated value. 

What are the other user configs which are overridden always?

Reply:
Add 10 as a constant value macro at the top.
Something like, CLUSTERMAN_OBSERVED_SCALING_TIME.

> it gets to 15min pretty quickly with the right amount of executors anyway.
Not sure, I follow this.